src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastCache.cs

```diff
@@ -250,7 +250,7 @@ internal static CastResult TryGet(int[] table, nuint source, nuint target)
             TableMask(ref tableData) = size - 1;
 
             // Fibonacci hash reduces the value into desired range by shifting right by the number of leading zeroes in 'size-1'
-            byte shift = (byte)BitOperations.LeadingZeroCount(size - 1);
+            byte shift = (byte)BitOperations.LeadingZeroCount((ulong)(size - 1));
```

Comment:
Out of curiosity, why ulong instead of uint?

Reply:
Originally this was:
```c
    // Fibonacci hash reduces the value into desired range by shifting right by the number of leading zeroes in 'size-1'
    DWORD bitCnt;
#if HOST_64BIT
    BitScanReverse64(&bitCnt, size - 1);
    HashShift(tableData) = (BYTE)(63 - bitCnt);
#else
    BitScanReverse(&bitCnt, size - 1);
    HashShift(tableData) = (BYTE)(31 - bitCnt);
#endif
```

Reply:
I think this should be `ulong` on 64bit and `uint` on 32bit.

Reply:
`size` is int, so `uint` makes the most sense here. Alternatively, this can call `Int32.LeadingZeroCount` that would make the cast unnecessary.

Reply:
Just a bit more explanation for `ulong` vs. `uint` differences.
When reducing the hashed value into an array index we want to take the top N bits, where N is the nonzero bits in `size-1`.  
I.E. For the `size = 1024`, the shift will be `53` on 64 bit and `21` on 32 bit.

It is also possible to just use `uint` and add `32` to the result on 64bit.

Reply:
Ah, ok. I see why this should be `nuint`.

Does it mean that this is actually fixing a functional regression? The number of leading bits is wrong with Int128 that this ends up calling currently.

Reply:
> Does it mean that this is actually fixing a functional regression?

This code is not on a hot path (shift is computed on table resize, which is rare), so 128bit math has no perf impact.
What saves us on the correctness side is that in C# bitwise shift masks the count according to the operand size, so we end up using `modulo 63` value.  
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/bitwise-and-shift-operators#shift-count-of-the-shift-operators

Basically - this ends up working correctly, but not intentionally.